src/main/java/blf/core/writers/XesWriter.java

```diff
@@ -174,38 +161,32 @@ public void addStringList(@NonNull String key, @NonNull List<String> values) {
     }
 
     @Override
-    protected void writeState(String filenameSuffix) throws Throwable {
+    protected void writeState(String fileNameSuffix) {
         LOGGER.info("Xes export started.");
-        // Before
-        Map<String, Map<String, XTrace>> formerTraces = deepCopyTraces();
-        Map<String, Map<String, Map<String, XEvent>>> formerEvents = deepCopyEvents();
-        // Modification of state
         final Map<String, XLog> logs = this.getLogs();
-        // Restore of state
-        this.traces.clear();
-        this.events.clear();
-        this.traces.putAll(formerTraces);
-        this.events.putAll(formerEvents);
         try {
             final File folder = this.getOutputFolder().toAbsolutePath().toFile();
             final XesXmlSerializer serializer = new XesXmlSerializer();
             for (Entry<String, XLog> entry : logs.entrySet()) {
-                final String filename = String.format("log_%s_%s.xes", entry.getKey(), filenameSuffix);
+                final String filename = String.format("log_%s_%s.xes", entry.getKey(), fileNameSuffix);
                 final File file = new File(folder, filename);
-                try (final FileOutputStream outputStream = new FileOutputStream(file);) {
+                try (final FileOutputStream outputStream = new FileOutputStream(file)) {
                     serializer.serialize(entry.getValue(), outputStream);
                 }
             }
-        } catch (Exception t) {
-            LOGGER.info("Xes export finished unsuccessfully.");
-            final String message = "Error exporting data to XES.";
-            throw new ProgramException(message, t);
+        } catch (Exception e) {
+            final String errorMsg = "Error exporting data to XES.";
+            this.exceptionHandler.handleException(errorMsg, e);
+        } finally {
+            this.events.clear();
+            this.traces.clear();
+            this.element = null;
```

Comment:
The problem with clearing the state in finally is the following: writeState() is called multiple times in the new modes. But in safe batching mode, the state must not be cleared after the write. Otherwise we lose information in the output file.

Reply:
Removed